### PR TITLE
No if xfree

### PR DESCRIPTION
**Pull request by bandi13**
Remove 'if(ptr)' check before 'XFREE(ptr)'

### CODE REVIEW

**Comment by bandi13** [comment]
retest this please. Just a random test failure:
```
Client message: hello wolfssl!
I hear you fa shizzle!
trying server command line[1193]: SuiteTest -v 4 -l TLS13-AES128-GCM-SHA256 -2 -p 0 
repeating test without extended master secret
trying client command line[1194]: SuiteTest -v 4 -l TLS13-AES128-GCM-SHA256 -F 1 -2 -n -p 36085 
FAIL scripts/unit.test (exit status: 1)
<<<
Error: Test runtime=91sec. Make Check RESULT = 2
Exiting with status: 1
Script ran for 269 seconds.
script returned exit code 1
```

**Review by dgarske — CHANGES_REQUESTED** [review]
Please make sure all XFREE macros definitions have the `if()` check and clearly document this in that area of the headers. Its already a rule and I'm glad we are getting consistent with this, but let's document all `#define XFREE` examples.

**Review by dgarske — APPROVED** [review]

File: wolfcrypt/src/sp_arm32.c [hunk 1]
```diff
@@ -12027,8 +12027,7 @@ static int sp_2048_mod_exp_32(sp_digit* r, const sp_digit* a, const sp_digit* e,
     }
 
 #ifdef WOLFSSL_SP_SMALL_STACK
-    if (td != NULL)
-        XFREE(td, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    XFREE(td, NULL, DYNAMIC_TYPE_TMP_BUFFER);
```
Comment (SparkiDev) [inline wolfcrypt/src/sp_arm32.c:12030]: This code is generated.
I'll have to go back and change the scripts to produce this code.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Review by douzzer — CHANGES_REQUESTED** [review]
`quantum-safe-wolfssl-all-clang-tidy` found something to complain about:
```
[quantum-safe-wolfssl-all-clang-tidy] [22 of 32] [a31d8c5ce7]
    configure...   real 0m26.333s  user 0m11.293s  sys 0m16.875s
    build...d350ba6c41 (<andras@wolfssl.com> 2024-08-06 10:44:59 -0400 7938)     XFREE(s1, key->heap, DYNAMIC_TYPE_DILITHIUM);
/tmp/tmp.4346_12509/wolfssl_test_workdir.35142/wolfssl/wolfcrypt/src/dilithium.c:7938:5: warning: Access to field 'heap' results in a dereference of a null pointer (loaded from variable 'key') [clang-analyzer-core.NullDereference]
7938 |     XFREE(s1, key->heap, DYNAMIC_TYPE_DILITHIUM);
|     ^
[...]
```

**Review by douzzer — CHANGES_REQUESTED** [review]
`clang-tidy-asn-template-sp-all-small-stack` found a similar defect in `rsa.c`:
```
[clang-tidy-asn-template-sp-all-small-stack] [32 of 32] [a31d8c5ce7]
    configure...   real 0m26.067s  user 0m11.164s  sys 0m16.735s
    build...d350ba6c41 (<andras@wolfssl.com> 2024-08-06 10:44:59 -0400 5084)     XFREE(p, key->heap, DYNAMIC_TYPE_RSA);
/tmp/tmp.4346_12509/wolfssl_test_workdir.35142/wolfssl/wolfcrypt/src/rsa.c:5084:5: warning: Access to field 'heap' results in a dereference of a null pointer (loaded from variable 'key') [clang-analyzer-core.NullDereference]
5084 |     XFREE(p, key->heap, DYNAMIC_TYPE_RSA);
|     ^
[...]
```

**Review by douzzer — APPROVED** [review]

**Pull request merged**